build.rs

```diff
@@ -134,10 +185,180 @@ fn fix_config_map(mut config_map: HashMap<String, String>) -> HashMap<String, St
+            }
+        }
+    }
 
+    Ok(entries)
+}
+
+/// Find cross compilation information from sysconfigdata file
+///
+/// first find sysconfigdata file which follows the pattern [`_sysconfigdata_{abi}_{platform}_{multiarch}`][1]
+///
+/// The ABI flags can be either u, d, or m according to PEP3148. Default flags became empty since
+/// m was removed in python 3.8. (?:u|d|m|) a non capturing group for these flags
+///
+/// platform follows the output from [sys.platform][2]
+/// [a-z0-9]+
+///
+/// Multi-arch is the target triple. ([a-z_\-0-9]*)? capturing group which might not be present.
+///
+/// # Examples
+/// ```txt
+/// _sysconfigdata__freebsd_.py
+/// _sysconfigdata_m_linux_x86_64-linux-gnu.py
+/// _sysconfigdata_d_darwin_x86_64-apple-darwin.py
+/// _sysconfigdata_u_windows_i686-pc-windows-gnu.py
+/// ```
+///
+/// [1]: https://github.com/python/cpython/blob/3.8/Lib/sysconfig.py#L348
+/// [2]: https://docs.python.org/3/library/sys.html#sys.platform
+#[cfg(feature = "cross-compile")]
+fn load_cross_compile_from_sysconfigdata(
+    python_paths: PythonPaths,
+) -> Result<(InterpreterConfig, HashMap<String, String>)> {
+    let sysconfig_re =
+        regex::Regex::new(r"_sysconfigdata_(?:u|d|m|)_[a-z0-9]+_([a-z_\-0-9]*)?\.py$")?;
+    let mut walker = walkdir::WalkDir::new(&python_paths.lib_dir).into_iter();
+    let sysconfig_path = loop {
+        let entry = match walker.next() {
+            Some(Ok(entry)) => entry,
+            None => bail!("Could not find sysconfigdata file"),
+            _ => continue,
+        };
+        let entry = entry.into_path();
+        if sysconfig_re.is_match(entry.to_str().unwrap()) {
+            break entry;
+        }
+    };
+    let config_map = parse_sysconfigdata(sysconfig_path)?;
+
+    let shared = match config_map
+        .get("Py_ENABLE_SHARED")
+        .map(|x| x.as_str())
+        .ok_or("Py_ENABLE_SHARED is not defined")?
+    {
+        "1" | "true" | "True" => true,
+        "0" | "false" | "False" => false,
+        _ => panic!("Py_ENABLE_SHARED must be a bool (1/true/True or 0/false/False"),
+    };
+
+    let (major, minor) = match config_map.get("VERSION") {
+        Some(s) => {
+            let split = s.split('.').collect::<Vec<&str>>();
+            (split[0].parse::<u8>()?, split[1].parse::<u8>()?)
+        }
+        None => bail!("Could not find python version"),
+    };
+
+    let ld_version = match config_map.get("LDVERSION") {
+        Some(s) => s.clone(),
+        None => format!("{}.{}", major, minor),
+    };
+    let python_version = PythonVersion {
+        major,
+        minor: Some(minor),
+        implementation: PythonInterpreterKind::CPython,
+    };
+
+    let interpreter_config = InterpreterConfig {
+        version: python_version,
+        libdir: Some(python_paths.lib_dir.to_owned()),
+        shared,
+        ld_version,
+        base_prefix: "".to_string(),
+        executable: PathBuf::new(),
+        calcsize_pointer: None,
```

Comment:
It looks like we can use `SIZEOF_VOID_P` for this.

Reply:
applied